Src/GoogleApis.Auth.DotNet4/OAuth2/ServiceAccountCredential.cs

```diff
@@ -72,6 +76,20 @@ public Initializer(string id, string tokenServerUrl) : base(tokenServerUrl)
                 Scopes = new List<string>();
             }
 
+            /// <summary>Constructs a new initializer using the given id and the private key in the ClientCredentialParameters.</summary>
+            public Initializer(ClientCredentialParameters clientCredentialParameters)

```

Comment:
What this takes as input is not a JsonKey. It takes a CredentialParameters type and hence I feel the constructor would be a better suited here. However if you strong feel that this is what you prefer, may I suggest another option, which is to use FromCredentialParameters() instead of FromJsonKey().